scripts/test/run-tests.sh

```
@@ -53,12 +50,12 @@ convert_and_test "qemux86_64" \
   "${UBUNTU_IMAGE}.gz" \
   "configs/qemux86-64_config" || test_result=$?
 
-convert_and_test "raspberrypi" \
-  "release-1" \
-  "${RASPBIAN_IMAGE_URL}" \
-  "${RASPBIAN_IMAGE}.img" \
-  "${RASPBIAN_IMAGE}.zip" \
-  "configs/raspberrypi3_config" || test_result=$?
+if [ -f deploy/raspberrypi-mender-raspbian.sdimg ]; then
+  run_tests "raspberrypi" "mender-raspbian" || test_result=$?
+else
+  echo "FAILED! This test needs a pre-converted Raspbian image. See the convert_raspbian job in the .gitlab-ci.yml file for how to generate, and then put the *uncompressed* image in the deploy folder"
+  test_result=1
+fi
```

Not sure what is going on here. Why do we need pre-converted image here?

---

We want to test the exact image we are releasing, which is made in a different stage, and not using the `convert_and_test` call. This was mostly to give some heads up if people run this locally, though I don't expect many people to do that. Does that explain it?

I could wrap it in a special flag which we only call from `.gitlab-ci.yml`, to keep local tests passing, but I opted for being verbose about it instead. Let me know if you would like it changed.

---

Understood.

Probably not many people will try to run the tests locally, but at least I have appreciated to be able to do so and if not to much trouble it would be nice to continue doing it :)

---

While working on the tool I also thought it was of great value to build/test locally, so I think we should keep it.

In this script, I would do if the pre-converted image exists, run_tests, else, convert_and_test (as it was before).

---

> In this script, I would do if the pre-converted image exists, run_tests, else, convert_and_test (as it was before).

I don't want to use a method that will silently pass if the image is missing (risk of false negative). If it's desired to convert using the old method, then I will base that on a flag instead (`--use-prebuild-image` or something), and still fail if the flag is specified and the image is missing.

---

Good point about the false positive. I like the extra flag suggestion, it is not that hard to implement right?

Maybe we just actually run the tests on this new prebuild image in addition to the build+test one, else we also risk the latter to fail the conversion or something down the road (for example if there is an problem with the link upstream or something)

---

Indeed, this is what I ended up doing. The script now requires a parameter, either `--all` or `--prebuilt-image ...`. So the prebuilt image is now an additional test, not a replacement test.